command/agent_test.go

```diff
@@ -1438,7 +1438,114 @@ auto_auth {
 	return config, cleanup
 }
 
-func TestAgent_Cache_Retry(t *testing.T) {
+func TestAgent_Cache_DynamicSecret(t *testing.T) {
+	logger := logging.NewVaultLogger(hclog.Trace)
+	coreConfig := &vault.CoreConfig{
+		Logger: logger,
+		CredentialBackends: map[string]logical.Factory{
+			"approle": credAppRole.Factory,
+		},
+	}
+	cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+	})
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	serverClient := cluster.Cores[0].Client
+
+	// Unset the environment variable so that agent picks up the right test
```

Comment:
I know other agent tests do this, but I think it's misguided and we shouldn't.

Reply:
You're right - it works without, so I'll remove